cli/spcli/actor.go

```diff
@@ -34,6 +36,112 @@ import (
+		Name:      "settle-deal",
+		Usage:     "Settle deals manually, if dealIds are not provided all deals will be settled",
+		ArgsUsage: "[...dealIds]",
+		Flags: []cli.Flag{
+			&cli.IntFlag{
+				Name:  "confidence",
+				Usage: "number of block confirmations to wait for",
+				Value: int(buildconstants.MessageConfidence),
+			},
+		},
+
+		Action: func(cctx *cli.Context) error {
+			api, acloser, err := lcli.GetFullNodeAPIV1(cctx)
+			if err != nil {
+				return err
+			}
+			defer acloser()
+
+			ctx := lcli.ReqContext(cctx)
+			maddr, err := getActor(cctx)
+			if err != nil {
+				return err
+			}
+
+			var (
+				dealIDs []uint64
+				dealId  uint64
+			)
+
+			// if no deal ids are provided, get all the deals for the miner
+			if dealsIdArgs := cctx.Args().Slice(); len(dealsIdArgs) != 0 {
+				for _, d := range dealsIdArgs {
+					dealId, err = strconv.ParseUint(d, 10, 64)
+					if err != nil {
+						return xerrors.Errorf("Error parsing deal id: %w", err)
+					}
+
+					dealIDs = append(dealIDs, dealId)
+				}
+			} else {
+				if dealIDs, err = GetMinerAllDeals(ctx, api, maddr, types.EmptyTSK); err != nil {
+					return xerrors.Errorf("Error getting all deals for miner: %w", err)
+				}
+			}
+
+			mi, err := api.StateMinerInfo(ctx, maddr, types.EmptyTSK)
+			if err != nil {
+				return xerrors.Errorf("Error getting miner info: %w", err)
+			}
+
+			dealParams := bitfield.NewFromSet(dealIDs)
+			params, err := actors.SerializeParams(&dealParams)
+			if err != nil {
+				return err
+			}
+
+			smsg, err := api.MpoolPushMessage(ctx, &types.Message{
+				To:     maddr,
+				From:   mi.Owner,
+				Value:  types.NewInt(0),
+				Method: marketactor.Methods.SettleDealPaymentsExported,
+				Params: params,
+			}, nil)
+			if err != nil {
+				return err
+			}
+
+			res := smsg.Cid()
+
+			fmt.Printf("Requested deal settlement in message %s\nwaiting for it to be included in a block..\n", res)
+
+			// wait for it to get mined into a block
+			wait, err := api.StateWaitMsg(ctx, res, uint64(cctx.Int("confidence")), lapi.LookbackNoLimit, true)
+			if err != nil {
+				return xerrors.Errorf("Timeout waiting for deal settlement message %s", res)
+			}
+
+			if wait.Receipt.ExitCode.IsError() {
+				return xerrors.Errorf("Failed to execute withdrawal message %s: %w", wait.Message, wait.Receipt.ExitCode.Error())
+			}
+
+			nv, err := api.StateNetworkVersion(ctx, wait.TipSet)
+			if err != nil {
+				return err
+			}
+
+			if nv >= network.Version14 {
+				var settlementReturn markettypes14.SettleDealPaymentsReturn
+				if err = settlementReturn.UnmarshalCBOR(bytes.NewReader(wait.Receipt.Return)); err != nil {
+					return err
+				}
+
+				// if all the deals were successful
+				if settlementReturn.Results.SuccessCount == uint64(len(dealIDs)) {
+					fmt.Printf("Successfully settled %d deals\n", settlementReturn.Results.SuccessCount)
+				} else {
+					fmt.Printf("Settled %d out of %d deals\n", settlementReturn.Results.FailCount, len(dealIDs))
```

Comment:
Incorrect success count calculation.

```suggestion
					fmt.Printf("Settled %d out of %d deals\n", len(dealIDs) - settlementReturn.Results.FailCount, len(dealIDs))
```

Reply:
Done.

Just a question is there any other state the deal could be in apart from `success` and `fail`?
If not then we can do something like this: 
`fmt.Printf("Settled %d out of %d deals\n", settlementReturn.Results.SuccessCount, len(dealIDs))`

Reply:
Return type is already predefined so it will always stick to those types. Your suggestion seems better than current one.

Reply:
I will approve once this final change is finished.

Reply:
Done.